libraries/Microsoft.Bot.Builder.Dialogs/DialogManager.cs

```diff
@@ -361,6 +363,24 @@ private async Task<DialogTurnResult> HandleSkillOnTurnAsync(DialogContext dc, Ca
             return turnResult;
         }
 
+        /// <summary>
+        /// Recursively traverses the <see cref="Dialog"/> tree and registers instances of <see cref="DialogContainer"/>
+        /// in the <see cref="DialogSet"/> for this <see cref="DialogManager"/> instance.
+        /// </summary>
+        /// <param name="dialog">Root of the <see cref="Dialog"/> subtree to iterate and register containers from.</param>
+        private void RegisterContainerDialogs(Dialog dialog)
+        {
+            if (dialog is DialogContainer container)
+            {
+                Dialogs.Add(container);
```

Comment:
>Dialogs.Add(container); [](start = 16, length = 23)

I think this might add a duplicate, unless DialogSet protects against that.

Reply:
It does protect but still we don't want to do stuff we don't need to do. I'll fix.